test/cql-pytest/test_describe.py

```diff
@@ -0,0 +1,166 @@
+
+    with new_test_table(cql, test_keyspace, "id int primary key") as table:
+        table_name = table.split(".")[1]
+        cql.execute(f"use {test_keyspace}")
+        ks_result = cql.execute("DESC TABLES")
+        ks_names = extract_names(ks_result)
+
+        assert "local" not in ks_names
+        assert "sstable_activity" not in ks_names
+        assert table_name in ks_names
+
+# Test that `DESC TABLE {name}` contains create statement for table
+def test_desc_table(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "id int primary key") as table:
+        table_name = table.split(".")[1]
+        result = cql.execute(f"DESC TABLE {test_keyspace}.{table_name}")
+        desc = extract_describe(result)
+
+        assert f"CREATE TABLE {test_keyspace}.{table_name}" in desc
+        assert "id int" in desc
+        assert "PRIMARY KEY (id)" in desc
+
+# Test that `DESC TYPES` contains newly created user type
+def test_desc_types(cql, test_keyspace):
+    with new_type(cql, test_keyspace, "(a int)") as typ:
+        typ_name = typ.split(".")[1]
+        cql.execute(f"USE {test_keyspace}")
+        result = cql.execute("DESC TYPES")
+        names = extract_names(result)
+
+        assert typ_name in names
+
+# Test that `DESC TYPE {name}` contains create statement for user type
+def test_desc_type(cql, test_keyspace):
+    type_desc = "a int"
+    with new_type(cql, test_keyspace, f"({type_desc})") as typ:
+        result = cql.execute(f"DESC TYPE {typ}")
+        desc = extract_describe(result)
+
+        assert f"CREATE TYPE {typ}" in desc
+        assert type_desc in desc
+
+# Test that `DESC FUNCTIONS` contains newly created user function
+def test_desc_functions(scylla_only, cql, test_keyspace):
+    fn_schema = "(val int)"
+    fn_null = "RETURNS NULL ON NULL INPUT"
+    fn_return = "RETURNS int"
+    fn_language = "LANGUAGE lua"
+    fn_body = "AS 'return -val'"
+    with new_function(cql, test_keyspace, f"{fn_schema} {fn_null} {fn_return} {fn_language} {fn_body}") as fn:
+        cql.execute(f"USE {test_keyspace}")
+        result = cql.execute("DESC FUNCTIONS")
+        names = extract_names(result)
+
+        assert fn in names
+
+# Test that `DESC FUNCTION {name}` contains create statement for user function
+def test_desc_function(scylla_only, cql, test_keyspace):
+    fn_schema = "(val int)"
+    fn_null = "RETURNS NULL ON NULL INPUT"
+    fn_return = "RETURNS int"
+    fn_language = "LANGUAGE lua"
+    fn_body = "return -val"
+    with new_function(cql, test_keyspace, f"{fn_schema} {fn_null} {fn_return} {fn_language} AS '{fn_body}'") as fn:
+        result = cql.execute(f"DESC FUNCTION {test_keyspace}.{fn}")
+        desc = extract_describe(result)
+
+        assert f"CREATE FUNCTION {test_keyspace}.{fn}" in desc
+        assert fn_schema in desc
+        assert fn_null in desc
+        assert fn_return in desc
+        assert fn_language in desc
+        assert fn_body in desc
```

Comment:
I think it's better to compare the full description in one "in desc" instead of all these pieces. What if the description contain all the pieces but in the wrong order, or with other junk in the middle? This test will miss it.